packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,134 @@
+"use client";
+
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { twMerge } from "tailwind-merge";
+
+import { FREQUENCY_OPTIONS } from "@/models";
+import { BodyText, RadioButton, TextInput } from "@/ui";
+
+interface FrequencyOptionsCardProps {
+  frequency: FREQUENCY_OPTIONS;
+  setEndDate: (date: Date) => void;
+}
+
+const defaultFrequencyOptions = {
+  [FREQUENCY_OPTIONS.hour]: ["6", "12", "24"],
+  [FREQUENCY_OPTIONS.day]: ["7", "15", "30"],
+  [FREQUENCY_OPTIONS.week]: ["4", "8", "12"],
+  [FREQUENCY_OPTIONS.month]: ["2", "6", "12"],
+};
+const maxCustomFrequencies = {
+  [FREQUENCY_OPTIONS.hour]: 96,
+  [FREQUENCY_OPTIONS.day]: 365,
+  [FREQUENCY_OPTIONS.week]: 52,
+  [FREQUENCY_OPTIONS.month]: 24,
+};
+const postiveIntegerOnly = /^$|^[0-9]+$/;
+
+const getEndDateByDefaultFrequency = (
+  frequency: FREQUENCY_OPTIONS,
+  timeAmount: number
+) => {
+  const dateNow = new Date();
+  const endDateByDefaultFrequency = {
+    [FREQUENCY_OPTIONS.hour]: dateNow.setHours(dateNow.getHours() + timeAmount),
+    [FREQUENCY_OPTIONS.day]: dateNow.setDate(dateNow.getDate() + timeAmount),
+    [FREQUENCY_OPTIONS.week]: dateNow.setDate(
+      dateNow.getDate() + timeAmount * 7
+    ),
+    [FREQUENCY_OPTIONS.month]: dateNow.setMonth(
+      dateNow.getMonth() + timeAmount
+    ),
+  };
+
+  return new Date(endDateByDefaultFrequency[frequency]);
+};
+
+const getCroppedFrequency = (frequency: FREQUENCY_OPTIONS) => {
+  const isMonthFrequency = frequency === FREQUENCY_OPTIONS.month;
+
+  return isMonthFrequency ? frequency.substring(0, 2) : frequency.charAt(0);
+};
+
+export const FrequencyOptionsCard = ({
+  frequency,
+  setEndDate,
+}: FrequencyOptionsCardProps) => {
+  const [defaultFrequency, setDefaultFrequency] = useState(
+    defaultFrequencyOptions[frequency][0]
+  );
+  const [customFrequency, setCustomFrequency] = useState("");
+  const [placeholder, setPlaceholder] = useState("");
+
+  const handleCustomFrequencyChange = (
+    event: ChangeEvent<HTMLInputElement>
+  ) => {
+    const newValue = event.target.value;
+    const isWithinRange = Number(newValue) <= maxCustomFrequencies[frequency];
+
+    if (postiveIntegerOnly.test(newValue) && isWithinRange) {
+      setDefaultFrequency("");
+      setCustomFrequency(newValue);
+    }
+  };
+
+  useEffect(() => {
+    const placeholderText = `Custom ${frequency} (max: ${
+      maxCustomFrequencies[frequency]
+    } ${getCroppedFrequency(frequency)})`;
```

Comment:
<img width="207" alt="image" src="https://github.com/user-attachments/assets/f38792cc-64d7-4bc7-9da2-f248a5e6bc7b">

I would remove the "d"/"w"/"m" from the placeholder. In this case would be (max: 52)

cause it might imply I have to put it, I prefer just the number.

```suggestion
    const placeholderText = `Custom ${frequency} (max: ${
      maxCustomFrequencies[frequency]
    } )`;
```